source/crud/tests/unified/estimatedDocumentCount.yml

```diff
@@ -0,0 +1,276 @@
+  - collection:
+      id: &collection1 collection1
+      database: *database0
+      collectionName: &collection1Name coll1
+
+initialData:
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents:
+      - { _id: 1, x: 11 }
+      - { _id: 2, x: 22 }
+      - { _id: 3, x: 33 }
+
+tests:
+  - description: "estimatedDocumentCount uses $collStats on 4.9.0 or greater"
+    runOnRequirements:
+      - minServerVersion: "4.9.0"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection0Name
+                pipeline: &pipeline
+                  - $collStats: { count: {} }
+                  - $group: {
+                    _id: 1,
+                    n: { $sum: $count },
+                  }
+              commandName: aggregate
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount with maxTimeMS on 4.9.0 or greater"
+    runOnRequirements:
+      - minServerVersion: "4.9.0"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        arguments:
+          maxTimeMS: 6000
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection0Name
+                pipeline: &pipeline
+                  - $collStats: { count: {} }
+                  - $group: {
+                    _id: 1,
+                    n: { $sum: $count },
+                  }
+                maxTimeMS: 6000
+              commandName: aggregate
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount on non-existent collection on 4.9.0 or greater"
+    runOnRequirements:
+      - minServerVersion: "4.9.0"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection1
+        expectResult: 0
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection1Name
+                pipeline: &pipeline
+                  - $collStats: { count: {} }
+                  - $group: {
+                    _id: 1,
+                    n: { $sum: $count },
+                  }
+              commandName: aggregate
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount errors correctly on 4.9.0 or greater--command error"
+    runOnRequirements:
+      - minServerVersion: "4.9.0"
+    operations:
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: [ aggregate ]
+              errorCode: 11000
```

Comment:
When specifying an `errorCode` in a test file, please add a comment with the readable name. Quoting [expectedError](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#expectederror):

> YAML files SHOULD use a comment to note the corresponding code name (e.g. `errorCode: 26 # NamespaceNotFound`).

That was written with `expectError` in mind but I think it's good practice everywhere, especially if the name isn't already in the test description.

I realize it's only important that the error code here is not retryable, but since 11000 is DuplicateKey (write related) I think it'd be clearer to use something more generic. Looking through [error_codes.yml](https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.yml), InternalError(1) or UnknownError(8) seem preferable (neither is retryable).

Reply:
That sounds good! I've switched to `UnknownError(8)` and left a comment at each point in the yaml.